docs/architecture/architecture.md

```diff
@@ -0,0 +1,671 @@
+[virtcontainers project](https://github.com/containers/virtcontainers), which
+provides a generic, runtime-specification agnostic, hardware-virtualized containers
+library.
+
+### Configuration
+
+The runtime uses a TOML format configuration file called `configuration.toml`. By
+default this file is installed in the `/etc/clear-containers` directory.
+Most users will not need to modify the configuration file.
+
+The file is well commented and provides a few "knobs" that can modify the
+behaviour of the runtime.
+
+The configuration file is also used to enable runtime debug output (see
+https://github.com/clearcontainers/runtime#debugging).
+
+### Significant OCI commands
+
+Here we will describe how `cc-runtime` handles the most important OCI commands.
+
+#### [`create`](https://github.com/clearcontainers/runtime/blob/master/create.go)
+
+When handling the OCI `create` command, `cc-runtime` goes through the following steps:
+
+1. Create the networking container namespace on the host, according to the container
+  OCI configuration file. We only support networking namespaces for now, but
+  will support more of them later.
+2. Run all the [prestart OCI hooks](https://github.com/opencontainers/runtime-spec/blob/master/config.md#hooks)
+  in the host namespaces created in step 1, as described by the OCI container
+  configuration file.
+3. [Set up the container networking namespace up](#networking). This is when
+  all networking interfaces are created and setup inside the namespace, according
+  to the selected pod networking model (CNM for Docker or CNI for Kubernetes).
+4. Create and start the virtual machine running the container process. The VM
+  will run inside the host namespaces created during step 1, and its `systemd`
+  instance will spawn the `cc-agent` daemon.
+5. Register the virtual machine with `cc-proxy`.
+6. The `cc-proxy` waits for the agent to signal that it is ready and then returns
+  a token. This token uniquely identifies a process within a container inside
+  the virtual machine.
+7. Spawn the `cc-shim` process providing two arguments:
+  `cc-shim --token $(token) --uri $(uri)`
+   * The proxy URL, which can be either a UNIX or a TCP socket.
+   * The token for the container process it needs to monitor.
+8. The `cc-shim` connects to the proxy and signals which container process it
+  is going to monitor by passing its token through the `cc-proxy` `connectShim`
+  command.
+
+![Docker create](arch-images/create.png)
+
+At this point, the virtual machine that will run the containers workloads
+is up and running, and the [`cc-agent`](#agent) is ready to process container
+life cycle commands. The pod inside the virtual machine is not created, and
+the containers are not running yet. This will be done through the OCI 
+[`start`](#start) command
+
+#### [`start`](https://github.com/clearcontainers/runtime/blob/master/start.go)
+
+With namespace containers `start` launches a traditional Linux container process
```

Comment:
I'd put a ',' in after containers -> 'With namespace containers, `start`...'  Will also be more consistent with the sentence that follows.

Reply:
done